### ISSUE TITLE
Wrapped sequence view

### ISSUE THREAD

**Issue by BevanR** [GMOD/jbrowse#823]
Does JBrowse support a wrapped view of sequences?

If not, is such a feature on the radar?  Would a contribution for it be well received?

**Comment by BevanR** [comment]
- Any advice, insights or thoughts on how to do it?
- Is there someone available to guide or mentor me if I take it on?

**Comment by erasche** [comment]
I would be very interested in seeing this implemented, we have a ton of circular phage genomes. :D

**Comment by BevanR** [comment]
I added a screenshot example to make it clearer.

**Comment by nathandunn** [comment]
@BevanR  Trying to clarify what you want here.  You are talking modifying the popup area under "Region Sequence" not the zoomed in sequence values? 

<img width="945" alt="screen shot 2016-11-14 at 10 02 22 am" src="https://cloud.githubusercontent.com/assets/751274/20276547/8bcfb902-aa51-11e6-82cf-e100a1b21b44.png">

**Comment by BevanR** [comment]
No.  I'm talking about the entire sequence.  Here is another example from Geneious.  This one is zoomed out most of the way:

![screen shot 2016-11-15 at 8 47 32 am](https://cloud.githubusercontent.com/assets/325176/20280236/8f2c0e10-ab10-11e6-92c6-5fb500d83751.png)

This is the same chromosome zoomed in a lot, but still in wrapped view;

![screen shot 2016-11-15 at 8 49 25 am](https://cloud.githubusercontent.com/assets/325176/20280232/8c940180-ab10-11e6-90b7-3fe43ad17322.png)

**Comment by nathandunn** [comment]
Thanks for the clarification.    We'll try to get back to you by the end of this week.

**Comment by enuggetry** [comment]
Sorry for the delayed response, I think I completely missed this thread.
We don't have anything like this on the roadmap, so we'd welcome the contribution.
In general, this looks like it wouldn't fit our existing view model (GenomeView class).  So, I think the extension would be a class that substitutes the GenomeView class.  You could take advantage of some of the location selection/zoom navigation.  Unfortunately, there is no built-in framework to handle a view selection of this sort, yet.  However, we do plan to do some work here.  @ihh has been working on a radial view called Rotunda https://github.com/gmod/jbrowse/tree/rotunda (a purely SVG/D3 view), which we plan in integrate with some form of semantical selection.  Thus, I see the framework opening up accommodate alternate genome views like yours.  

I think it's safe to say that you can implement in either Javascript framework, dojo or jquery, with GUI elements like dialog boxes leaning towards Bootstrap (not completely decided).  Implement your view to respect a DIV container.

**Comment by nathandunn** [comment]
Also, if you have an existing code-base in Java (if you are working within you're own Geneious plugin), you could use GWT to start building the component.  The newer GWT uses [JSInterop](http://www.luigibifulco.it/blog/en/blog/gwt-2-8-0-jsinterop) to do javascript integration.  That being said, I think forking off of  @ihh 's rotunda would be a better starting point if you are starting from scratch.
